File: 4107-find-missing-elements/find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        res = []
        st = set()
        mini = float('inf')
        maxi = float('-inf')
        for i in range(0,len(nums)):
            st.add(nums[i])
            if nums[i] < mini:
                mini = nums[i]
            if nums[i] > maxi:
                maxi = nums[i]

        for i in range(mini,maxi + 1):
            if i not in st:
                res.append(i)

        
        return res


                

File: 4107-find-missing-elements/find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        vector<int>result;
        unordered_set<int>st;
        int mini = INT_MAX;
        int maxi  = INT_MIN;
        for(int i=0;i<nums.size();i++){
            st.insert(nums[i]);
            if(nums[i] < mini){
                mini = nums[i];
            }
            if(nums[i] > maxi){
                maxi = nums[i];
            }
        }
        sort(nums.begin(),nums.end());
        int temp = 0;
        for(int i=mini;i<=maxi;i++){
            if(!st.count(i)){
                result.push_back(i);
            }
        }
        return result;
    }
};